### PR TITLE
CI: another master -> main fix

### PR DESCRIPTION
**Pull request by rgommers**
Follow-up to gh-18423

### CODE REVIEW

**Comment by rgommers** [comment]
GH Actions indeed fails immediately with the one PEP8 issue I introduced on purpose:

image

And so does Azure, so maybe the `master` to `main` in this PR isn't needed. @ganesh-k13 do you know what is up with that one?

image

**Comment by ganesh-k13** [comment]
Hi Ralf, I think there's a problem when we have a merge commit, `${{ github.base_ref }}` and `$(System.PullRequest.TargetBranch)
` is empty in case of merge commits, we also have to skip it in case of merge commits. But as mentioned here: https://github.com/numpy/numpy/pull/18423#pullrequestreview-593580994, seems like it does not work.

**Comment by rgommers** [comment]
From SciPy's Azure setup, this line handles it:
```
- job: Lint
  condition: and(succeeded(), ne(variables['Build.SourceBranch'], 'refs/heads/master'))  # skip for PR merges
```

**Comment by rgommers** [comment]
Somehow that doesn't work here, not sure why

**Comment by rgommers** [comment]
PR behavior is fine at least:

image

So let me revert the intentional PEP8 error, and try a merge again. It may be the `master` -> `main` here.

File: tools/linter.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def run_pycodestyle(self, diff):
         """
             Original Author: Josh Wilson (@person142)
             Source:
-              https://github.com/scipy/scipy/blob/master/tools/lint_diff.py
+              https://github.com/scipy/scipy/blob/main/tools/lint_diff.py
```
Comment (rgommers) [inline tools/linter.py:43]: this link will change very soon when SciPy renames its master branch to main, so just fixing up preemptively

**Comment by rgommers** [comment]
Lint jobs are green now, merging again.

**Comment by rgommers** [comment]
The one difference I can see with the SciPy setup is that SciPy doesn't have the `stage`, just top-level jobs:
```
- stage: InitialTests
  jobs:

  - job: Lint
```

**Comment by rgommers** [comment]
This PR did fix the Azure job on `main`:

image

So it's just GH Actions now, which SciPy doesn't have.

**Comment by ganesh-k13** [comment]
Oh @rgommers , it passes for Azure: https://dev.azure.com/numpy/numpy/_build/results?buildId=16031&view=logs&jobId=40b6d036-694f-5d1f-586c-a0e1fd689e6f&j=e5f5179f-20ee-5742-3c41-54a886f6822a, problem is with Github Actions

**Comment by ganesh-k13** [comment]
Should we try something like this: https://github.community/t/trigger-workflow-only-on-pull-request-merge/17359/3 but in reverse and use a `!`? Will look for cleaner ways meanwhile.

**Comment by rgommers** [comment]
Yep, was just sending a PR with `github.ref != 'refs/heads/main' `, I think that should work

**Comment by rgommers** [comment]
See gh-18567?

**Comment by ganesh-k13** [comment]
Thanks a lot for resolving it @rgommers , So sorry to have missed this.

**Pull request merged**